### DISCUSSION TITLE
Why Schmid Factor values are >0.5?

### DISCUSSION THREAD

**Discussion by Mustafa-Rifat** [mtex-toolbox/mtex#1641] [Ask Anything]
Hello,

I am calculating schmid factor of an EBSD map. The stress tensor I have is from a FEA model. I have normalized it before using. Also, to make sure my FEA model scratch direction (+x axis) matches with my EBSD scratch direction (+y axis), I rotated the normalized stress tensor CCW 90-deg. My code and result is attached here:
```
% crystal symmetry
CS = {... 
  crystalSymmetry('m-3m', [3.57 3.57 3.57], 'mineral', 'Ni', 'color', 'light blue'),...
  crystalSymmetry('m-3m', [3.57 3.57 3.57], 'mineral', 'Ni-superalloy', 'color', 'light blue')};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Matlab Directory';
 
% which files to be imported
fname = [pname '\Sample 2 postscratch Specimen 1 Site 1 Map Data 4.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = loadEBSD(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
grains_small_angle = calcGrains(ebsd,'angle',1*degree);
plot(ebsd,ebsd.orientations,'micronbar','off')
hold on
plot(grains.boundary,'linewidth',2)
hold on
plot(grains_small_angle.boundary,'linewidth',1)
hold off
sS = slipSystem.fcc(ebsd.CS)
sS = sS.symmetrise('antipodal');
%rotate the slip systems from crystal to specimen coordinate
sSLocal = grains.meanOrientation * sS;
%below one is Annealedd500nm_R20_mu0pt15_v2.odb stress tensor values from
%element 49657
stress_FEA=[8.12033E08 -6.75181E06 -1.99316E07;-6.75181E06 1.52078E08 -5.21907E07;-1.99316E07 -5.21907E07 -6.01906E07];
normalized_stress_FEA=stress_FEA./norm(stress_FEA,'fro');
%rotate the normalized_stress_FEA 90-deg CCW so that the FEA coordinate
%matches with the EBSD coordinate 
alpha=pi/2;
rotation_matrix=[cos(alpha) -sin(alpha) 0; sin(alpha) cos(alpha) 0; 0 0 1];
normalized_stress_FEA=rotation_matrix*normalized_stress_FEA*rotation_matrix';
sigma = stressTensor(normalized_stress_FEA);
SF = sSLocal.SchmidFactor(sigma);
[SFMax,active] = max(SF,[],2);
figure
plot(grains,SFMax,'micronbar','off','linewidth',2)
mtexColorbar 
sSactive = grains.meanOrientation .* sS(active);
hold on
% visualize the trace of the slip plane
quiver(grains,sSactive.trace,'color','b')

% and the slip direction
quiver(grains,sSactive.b,'color','r')
hold off
```
![image](https://user-images.githubusercontent.com/60717121/201181970-4efe69f5-04e8-4de7-8a5f-f383ee644ab3.png)

As you can see from the colorbar and I also calculated max(max(SF)), the maximum SF value is ~0.52. Additionally, I also observed negative SF values in the SF matrix. Why is this happening? Should not it be inside 0-0.5 range?

**Comment by kilir** [comment]
Hi,
reducing your example to
```
cs = crystalSymmetry('m-3m')
sS = slipSystem.fcc(cs);
sS = sS.symmetrise('antipodal');

o = equispacedSO3Grid(cs,'resolution',1*degree);

sSLocal = o * sS;

m=[8.12033E08 -6.75181E06 -1.99316E07;-6.75181E06 1.52078E08 -5.21907E07;-1.99316E07 -5.21907E07 -6.01906E07];

m=m./norm(m,'fro');
sigma = stressTensor(m);
SF = sSLocal.SchmidFactor(sigma);
max(SF(:))
```
I get 0.5.

Does that also happen for one of both phases?

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
I am considering only  one phase in my calculations (Ni superalloy). That is why I defined the ebsd=ebsd('Ni-superalloy'). What about getting negative values? I am getting so many negative values too in my SF matrix.

**Reply by kilir** [reply]
Ok.
Negative values are fine, they just have theopposite sense. If you don't care about that, just take the abs().

**Reply by Mustafa-Rifat** [reply]
Hi [Rüdiger Kilian](https://github.com/kilir),

I just copied your code and ran it. It is giving me max SF~0.5320>0.5. You can have a look at the image attached. I am not sure why this is happening, you had the similar issue?
![image](https://user-images.githubusercontent.com/60717121/201426734-e491b03c-5171-4177-844d-b6e67cbf866d.png)

**Reply by kilir** [reply]
Hi,
no, for me it tops at 0.5, nothing larger, using Matlab 2022b/2021b and the latest mtex development version.
Can you try with a simple stress tensor?
Cheers, Rüdiger

**Reply by Mustafa-Rifat** [reply]
I tried with simple stress tensors. It is showing maximum value as 0.5. Is there a reason for this? Image attached:
![image](https://user-images.githubusercontent.com/60717121/201485009-459d9064-9294-4348-b08f-247565002d39.png)

**Reply by kilir** [reply]
My guess would be, that on your system the stress tensor is not normalized to one, possibly by some rounding effect.
